fiona/fio/cat.py

```diff
@@ -64,26 +64,25 @@ def cat(ctx, files, precision, indent, compact, ignore_errors, dst_crs,
             layer[str(i)] = [0]
 
     try:
-        with ctx.obj['env']:
-            if bbox:
-                try:
-                    bbox = tuple(map(float, bbox.split(',')))
-                except ValueError:
-                    bbox = json.loads(bbox)
-            for i, path in enumerate(files, 1):
-                for lyr in layer[str(i)]:
-                    with fiona.open(path, layer=lyr) as src:
-                        for i, feat in src.items(bbox=bbox):
-                            if dst_crs or precision >= 0:
-                                g = transform_geom(
-                                    src.crs, dst_crs, feat['geometry'],
-                                    antimeridian_cutting=True,
-                                    precision=precision)
-                                feat['geometry'] = g
-                                feat['bbox'] = fiona.bounds(g)
-                            if use_rs:
-                                click.echo(u'\u001e', nl=False)
-                            click.echo(json.dumps(feat, **dump_kwds))
+        if bbox:
+            try:
+                bbox = tuple(map(float, bbox.split(',')))
+            except ValueError:
+                bbox = json.loads(bbox)
+        for i, path in enumerate(files, 1):
+            for lyr in layer[str(i)]:
+                with fiona.open(path, layer=lyr) as src:
+                    for i, feat in src.items(bbox=bbox):
+                        if dst_crs or precision >= 0:
+                            g = transform_geom(
+                                src.crs, dst_crs, feat['geometry'],
+                                antimeridian_cutting=True,
+                                precision=precision)
+                            feat['geometry'] = g
+                            feat['bbox'] = fiona.bounds(g)
+                        if use_rs:
+                            click.echo(u'\u001e', nl=False)
+                        click.echo(json.dumps(feat, **dump_kwds))
```

Comment:
Here we go: using the context manager removes a level of indentation and makes `ctx.obj['env']` an implementation detail.